packages/react-server/src/ReactFlightServer.js

```diff
@@ -4089,8 +4102,15 @@ export function halt(request: Request, reason: mixed): void {
     if (request.destination !== null) {
       flushCompletedChunks(request, request.destination);
     }
+    allReady(request);
   } catch (error) {
     logRecoverableError(request, error, null);
     fatalError(request, error);
   }
 }
+
+function allReady(request: Request) {
+  const onAllReady = request.onAllReady;
+  onAllReady();
+  request.onAllReady = noop;
+}
```

Comment:
We need to call this while aborting but we can also ping after so we ensure we only call this once

Reply:
This seems like a cop out. It makes me think about what else surrounding this function shouldn’t be called twice.

We so this for shell ready but that makes sense because the shell being ready doesn’t really have any semantic meaning (flushing it does) so it has no other implications but completing fully does have other implications, doesn’t it?

Reply:
Presumably it would if we could figure out resuming but no clear line of sight from me on that. I can rework it to not need this but it may be functionally equivalent. Like tracking whether we’ve marked the request as allReady. We could also only call already if we performed work and then hit zero abortable tasks but this complicates the implementation and I’m not sure it’s worth it if we’re end up not needing it